# Lab_7_sequence_comparison

This lab has **two parts**. 
**Part 1** The first part you will analyze the PGM sequences generated by everyone in the class last week. 
**Part 2** You will find a gene involved in the pathway of your choice


# PART 1

## Step 1 - Setup your lab 7 folder

As usual you will need to set up your new folder for lab 7. 
Make a directory called "lab_7"

## Step 2 - Copy the pgm1 sequences into your lab 7 folder

The sequences for lab are in a file called "all_pgm1.fasta".

Move into your lab 7 folder and copy ```/projects/class/binf3101_001/all_pgm1.fasta``` into your lab 7 folder

You will also need to copy in your PGM1 file from lab_6. 

At this point you should have the following files in your lab_7 folder:
- all_pgm1.fasta
- SRR00000.fasta


## Step 3 - Align the pgm1 sequences using MUSCLE

### Step 3a - load MUSCLE 
Muscle is already installed on the cluster. To load it you can use

```bash
module load muscle
```

### Step 3b - Align the whole class's PGM1 sequences

You will use MUSCLE to align the sequences in the file ```all_pgm1.fasta``` and save it to a new file called ```all_pgm1.align.fasta```

To learn how to use MUSCLE you can use the help documentation. There are two mandatory things you need to include and they are found under the **Basic usage** section of the help documentation. 

To access the help documentation for MUSCLE type

```bash
muscle --help
```

Once you generate the protein alignment you can take a look at it using the ```cat``` command 


## Step 4 - Create a consensus sequence

A consensus sequence looks at the alignment and reports the most common amino acid (or base pair) at that position. 

For example, if we had the alignment below

PLANK
PRANK
CRANK
PLANS

We could generate a consensus sequence that reports the consensus amino acid at a threshold of >50%
This would be:

PxANK

There is no consensus (x) for that second position because no amino acid reaches >50%

To generate our consensus sequence we are going to use the EMBOSS program **cons**

### Step 4a - Load emboss

To load emboss we will use

```ml emboss``

### Step 4b - Generate consensus Sequence

To generate the consensus sequence for all the PGM1 files (all_pgm1.align.fasta) we want to use the format 

```cons inputfile outputfile```

Fill in the input and output files to generate the consensus sequence. 

### Step 4c - Look at the consensus sequence

Take a look a the consensus sequence we generated and answer the questions below. 

Note - Some of the letters are upper-case and some are lower-case.

UPPERCASE letters = **high conservation** wherease 
lowercase letters = **low conservation**

# LQ 1

How many positions in the sequence aligment did not have a consesnsus amino acid?

# LQ 2

How many positions in the sequence alignment had low conservation for Threonine?

**_TIP_** In this case when we use grep we **do not use the -c command**. This is because it only counts **lines**.
If we want to match just one character (for example z) we would use grep with the ```-o``` option. For example the command would be
```bash 
grep -o "z" file.txt
z
z
z
z
```

That will print out every time there is a "z" in the file. 

If you don't want to count the number of times that is printed out we can add " | wc -l" to the end of our command. This calls the **w**ord **c**ount command and the ```-l``` option says to count the lines

```bash
grep -o "z" file.txt | wc -l
4
```

# LQ 3

How many positions in the sequence alignment had high conservation for Threonine?

## Step 5 - Compare your sequence to the consensus

To compare your PGM1 sequence (generated in lab 6) to the consensus sequence we are going to use **blastp**

In this case each file is on 1 sequence - so it doesn't matter which file we put as the query and which we put as the sequence. 

Save the blastp output to a new file so you can look at it and answer the questions below

# LQ 4 
Report the following statistic for your PGM1 vs the consensus
1. Score (bits)
2. E-value
3. Percent Identities
4. Percent Gaps






